### PR TITLE
backend/feat #623:thumbnail_addition_for_inapp_messages

### PR DESCRIPTION
**Pull request by AshishSaini25**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

want to add thumbnail for in app messages which will be png file in nature

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE atlas_driver_offer_bpp.transporter_config
+ADD COLUMN s3_media_file_url_pattern Text NOT NULL DEFAULT 'http://localhost:8016/ui/<DOMAIN>/s3media?filePath=<FILE_PATH>';
```
Comment (0utkarsh) [inline Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql:2]: Host can also be made a pram and be fetched from dhall in code, but yeah, this is fine and simple too. Just that the number of queries will be different in prod/master, we'll have to manually paste the host here during release.

File: Backend/lib/beckn-services/src/AWS/S3/Flow.hs [hunk 2]
```diff
@@ -150,6 +179,21 @@ mockPut baseDirectory bucketName path img =
...
+  (MonadIO m, Log m) =>
+  String ->
+  Text ->
+  String ->
+  ByteString ->
+  m ()
+mockPutPng baseDirectory bucketName path imageContent =
+  withLogTag "S3" $
+    liftIO $ do
+      let fullPath'Name = getFullPathMock baseDirectory bucketName path
```
Comment (0utkarsh) [inline Backend/lib/beckn-services/src/AWS/S3/Flow.hs:192]: Naming can be camel cased.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Message.hs [hunk 3]
```diff
@@ -101,17 +102,29 @@ createMediaEntry Common.AddLinkAsMedia {..} = do
...
-          & T.replace "<FILE_PATH>" filePath
-  _ <- fork "S3 put file" $ S3.put (T.unpack filePath) mediaFile
-  createMediaEntry Common.AddLinkAsMedia {url = fileUrl, fileType}
+  -- _ <- validateContentType
+  (publicUrl) <-
+    case fileType of
+      Common.S3Image -> do
+        mediaFile' <- L.runIO $ BS.readFile file
+        filePath <- createFilePath merchant.id.getId fileType ".png"
+        let publicUrl = generateFilePulicUrl transporterConfig.s3MediaFileUrlPattern filePath
```
Comment (0utkarsh) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Message.hs:113]: Typo. Also, brackets are redundant at some places

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql (hunk 1 above)
Comment (piyushKumar-1) [reply Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql:2]: +1

File: Backend/lib/beckn-services/src/AWS/S3/Flow.hs (hunk 2 above)
Comment (AshishSaini25) [reply Backend/lib/beckn-services/src/AWS/S3/Flow.hs:192]: ok

File: Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql (hunk 1 above)
Comment (AshishSaini25) [reply Backend/dev/migrations/dynamic-offer-driver-app/0155-add-s3_media_file_url_pattern.sql:2]: so how will I do this ?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Message.hs (hunk 3 above)
Comment (AshishSaini25) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Message.hs:113]: ok

File: Backend/lib/beckn-services/src/AWS/S3/Flow.hs [hunk 4]
```diff
@@ -132,6 +134,33 @@ put'' bucketName path img = withLogTag "S3" $ do
...
+  ) =>
+  Text ->
+  String ->
+  ByteString ->
+  m ()
+putPng'' bucketName path file = withLogTag "S3" $ do
+  let cmd = "aws s3api put-object --bucket " <> T.unpack bucketName <> " --key " <> path <> " --body " <> C.unpack file
+  liftIO $ callCommand cmd
+
+getPng'' ::
```
[the remark this answers is not in the document]
Comment (AshishSaini25) [reply Backend/lib/beckn-services/src/AWS/S3/Flow.hs:149]: yes, it was previous like that only , therefore I kept it like that

**Pull request closed without merging**